docs/osmlr_updates.md

```diff
@@ -0,0 +1,48 @@
+
+### Running the OSMLR Update Process
+
+The OSMLR generation process (osmlr) has been updated to accept an optional OSMLR pbf directory. THis indicates that the OSMLR generation process should apply updates based on the supplied OSMLR segment definitions. This triggers the segment "migration" process rather than creating an initial or new set of OSMLR segments.
+
+When updating or migrating OSMLR segments, the Valhalla data used by OSMLR generation must be "traffic-enabled". This means it must have OSMLR segment associations present in the Valhalla tiles (populated by running the valhalla_associate_segments process when creating Valhalla data). For new OSMLR segment generation the Valhalla tiles must not contain OSMLR segment associations.
+
+The OSMLR segment migration process essentially compares OSMLR segment definitions to Valhalla OSMLR association information to determine OSMLR segments that are still valid (have not changed significantly) and those that must be replaced. New segments are created both to replace OSMLR segments that have significant change as well as create new segments caused by new data (e.g. new roads).
+
+The OSMLR segment migration process proceeds as follows:
+
+* Iterate through OSMLR tiles (protocol buffer).
+* Copy all existing OSMLR segments to the new tile.
+* Identify all OSMLR segments that are associated to the Valhalla routing data. These segments are still valid and remain unchanged with their existing index within the tile remaining unchanged (OSMLR segments are immutable). Skip any previously deprecated OSMLR segments. What remains is the set of OSMLR segments that must be deprecated (they no longer associate to the routing data within specific tolerances). These segments are marked as deprecated/deleted with the date added to specify when the OSMLR segment was deprecated.
+* The OSMLR "creation" process is then run to generate new segments that either replace deprecated segments (if applicable) or identify newly added roads within the data. When creating new segments, any road segment that is already associated to an OSMLR is skipped / disallowed. New segments get sequential Ids within the tile, following all currently existing Ids so there is no collision of Ids within the tile. The new segments also have the date indicating when the segment was added to the OSMLR tile.
+
+As the protocol buffer output is updated, the GeoJSON output is similarly updated.
```

Comment:
As the protocol buffer output is updated, the GeoJSON output is similarly updated by removing the "deleted" or "superseded" features from the featurecollection.  New features are appended to the existing featurecollection for existing GeoJSON tiles.